hybrasyl/Objects/Creature.cs

```diff
@@ -330,42 +722,215 @@ public Creature LastHitter
+
+            if (castObject.Effects.Sound != null)
+                PlaySound(castObject.Effects.Sound.Id);
+
+            ActivityLogger.Info($"UseCastable: {Name} casting {castObject.Name}, {targets.Count()} targets");
+            foreach (var tar in targets)
+            {
+                if (castObject.Effects?.ScriptOverride == true)
+                {
+                    // TODO: handle castables with scripting
+                    // DoStuff();
+                    continue;
+                }
+                if (castObject.Effects?.Damage != null)
+                {
+                    Enums.Element attackElement;
+                    var damageOutput = NumberCruncher.CalculateDamage(castObject, tar, this);
+                    if (castObject.Element == Castables.Element.Random)
+                    {
+                        Random rnd = new Random();
+                        var Elements = Enum.GetValues(typeof(Enums.Element));
+                        attackElement = (Enums.Element)Elements.GetValue(rnd.Next(Elements.Length));
+                    }
+                    else if (castObject.Element != Castables.Element.None)
+                        attackElement = (Enums.Element)castObject.Element;
+                    else
+                        attackElement = (OffensiveElementOverride == Enums.Element.None ? OffensiveElementOverride : OffensiveElement);
+                    if (this is User) ActivityLogger.Info($"UseCastable: {Name} casting {castObject.Name} - target: {tar.Name} damage: {damageOutput}, element {attackElement}");
+
+                    tar.Damage(damageOutput.Amount, attackElement, damageOutput.Type, damageOutput.Flags, this, false);
+                    if (tar.Hp <= 0) { deadMobs.Add(tar); }
+                }
+                // Note that we ignore castables with both damage and healing effects present - one or the other.
+                // A future improvement might be to allow more complex effects.
+                else if (castObject.Effects.Heal != null)
+                {
+                    var healOutput = NumberCruncher.CalculateHeal(castObject, tar, this);
+                    tar.Heal(healOutput, this);
+                    if (this is User) ActivityLogger.Info($"UseCastable: {Name} casting {castObject.Name} - target: {tar.Name} healing: {healOutput}");
+                }
+
+                // Handle statuses
+
+                foreach (var status in castObject.Effects.Statuses.Add)
+                {
+                    Status applyStatus;
+                    if (World.WorldData.TryGetValueByIndex<Status>(status.Value, out applyStatus))
+                    {
+                        ActivityLogger.Info($"UseCastable: {Name} casting {castObject.Name} - applying status {status.Value}");
+                        ApplyStatus(new CreatureStatus(applyStatus, tar, castObject));
```

Comment:
Does duration / tick get passed through a different means? When I tried it, I only got the default.

Reply:
Override works now.